File: Problems/11. Container With Most Water.cpp
#include<vector>
#include<iostream>

using namespace std;



class Solution {
    public:
        int maxArea(vector<int>& height) {

            int i=0;
            int j = height.size()-1;

            int maxArea =0;
            int b;

            for(int i=0;i<height.size();i++){
                for(int j=i+1;j<height.size();j++){
                    int h = min(height[i],height[j]);
                    b = j-i;
                    maxArea = max(maxArea,h*breath);
                    
                }
            }
            
            return maxArea;
        }
};

int main(){

    Solution solution;

    vector<int> input = {1,8,6,2,5,4,8,3,7};

    int answer = solution.maxArea(input);

    cout << "Answer is: " << answer << endl;


    return 0;
}

File: 11. Container With Most Water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        result = -1

        def calculateWater(first,last):
            curHeight = min(height[first],height[last])
            curWidth = last - first
            return curHeight*curWidth

        first,last = 0,len(height)-1

        while first<last:
            result = max(calculateWater(first,last),result)
            if height[first] < height[last]:
                first += 1
            else:
                last -= 1
        
        return result

    # Simpler Code 
    def maxArea2023(self, height: List[int]) -> int:
        l, r, area = 0, len(height) - 1, float('-inf')
        while l < r:
            area = max(area, min(height[l],height[r])*(r-l))
            if height[l] < height[r]: l += 1
            else: r -= 1
        return area